### PR TITLE
Macros are stored in a collection

### PR DESCRIPTION
**Pull request by philreindl**
Move the macros into a collection.  The idea here is to set us up for having an arbitrary number of macros, rather than just 5.  The macro name is stored in the json, and can be used as the macro button label.  The description is stored, and is displayed as hover text.  Also, I moved the nightly build warning into the settings as well.

The changes in this pull request don't directly contribute much, but are setting the stage for better things to come.

### CODE REVIEW

File: src/com/willwinder/universalgcodesender/MainWindow.java [hunk 1]
```diff
@@ -2351,6 +2379,7 @@ private void setLocalLabels() {
         this.baudLabel.setText(Localization.getString("mainWindow.swing.baudLabel"));
         this.browseButton.setText(Localization.getString("mainWindow.swing.browseButton"));
         this.cancelButton.setText(Localization.getString("mainWindow.swing.cancelButton"));
```
Comment (winder) [inline src/com/willwinder/universalgcodesender/MainWindow.java]: I spy some debug code!

**Comment by winder** [comment]
This looks like a great step forward - I especially appreciate that you took the time to migrate legacy macros.

**Comment by winder** [comment]
I'm in the final stages of finishing #302, so I'm going to hold off on merging these changes until I get that branch stabilized. But don't let that stop you with the next few steps, getting rid of all those duplicate functions will be nice.

**Comment by philreindl** [comment]
Macro page is now rendered based on the macros that exist in the settings.  There is an extra, empty row to allow inserting new macros.  Since there's no add button, you need to close UGS and restart to get a new blank row.  Doesn't look like the panel is resizable, so (at least on my screen) you are currently limited to 8 macros.

It looks like the UI was originally built in NetBeans.  Is that the plan going forward?  Or is the ui code being manually maintained now?

**Comment by winder** [comment]
@philreindl Awesome! Most of the UI code is generated with Netbeans, including some of the code you changed here (there are block comments pointing out the sections that are auto-generated). I'm not planning to change this.

The way I'm handling custom GUI code by putting it into a subclassed component then adding that component with the Netbeans GUI builder. I'm hoping to finish the constant-memory work this week (fixing up the last of the unit tests), I'm happy to pull the GUI changes into something the Netbeans builder can use afterwards.

**Comment by philreindl** [comment]
Yeah I was afraid of that.  What had me questioning whether it was still in use is that the fonts and spacing don't match what I see in Netbeans.  Looks like there's some extra style being done outside the gui builder.  If you don't mind patching this up to be more netbeans friendly, that would be great.

**Comment by philreindl** [comment]
@winder OK I took a crack at adding a new MacroPanel component, which is installed to a JScrollPane via netbeans GUI builder.  It all appears to be working.

**Pull request merged**